From f74c92c1921bd2b79fd4450743f196fd05f86f16 Mon Sep 17 00:00:00 2001
From: Sibi Sankar <quic_sibis@quicinc.com>
Date: Wed, 29 Nov 2023 12:27:48 +0530
Subject: [PATCH 1/8] firmware: arm_scmi: Increase the maximum opp count in the
 perf protocol

The number of opps on certain variants of the X1E80100 SoC are greater
than current maximum of 16, so increase the MAX_OPP count to 32 (next
power of 2) to accommodate that.

Signed-off-by: Sibi Sankar <quic_sibis@quicinc.com>
Link: https://lore.kernel.org/r/20231129065748.19871-4-quic_sibis@quicinc.com
Reviewed-by: Cristian Marussi <cristian.marussi@arm.com>
Signed-off-by: Sudeep Holla <sudeep.holla@arm.com>
(cherry picked from commit c3f17d5f89fc72d7b170beaf393a8687ae938b19)
Signed-off-by: Grygorii Strashko <grygorii_strashko@epam.com>
---
 drivers/firmware/arm_scmi/perf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/firmware/arm_scmi/perf.c b/drivers/firmware/arm_scmi/perf.c
index dd344506b0a3..35d7194dabe0 100644
--- a/drivers/firmware/arm_scmi/perf.c
+++ b/drivers/firmware/arm_scmi/perf.c
@@ -24,7 +24,7 @@
 #include "protocols.h"
 #include "notify.h"
 
-#define MAX_OPPS		16
+#define MAX_OPPS		32
 
 enum scmi_performance_protocol_cmd {
 	PERF_DOMAIN_ATTRIBUTES = 0x3,
-- 
2.34.1

